File: 2163-kth-distinct-string-in-an-array/kth-distinct-string-in-an-array.ts
function kthDistinct(arr: string[], k: number): string {
    const counter = arr.reduce((acc, cur)=> {
        acc[cur] = (acc[cur] ?? 0) + 1;
        return acc
    }, {})
    return Object.keys(counter).filter(key=>counter[key] === 1)[k - 1] ?? ''
};

File: 2163-kth-distinct-string-in-an-array/kth-distinct-string-in-an-array.go
func kthDistinct(arr []string, k int) string {
    counter := make(map[string]int)
    counterOrder := []string{}
    for _, value := range arr {
        counter[value]++
        if counter[value] == 1 {
            counterOrder = append(counterOrder, value)
        }
    }
    kth := 1
    for _, key := range counterOrder {
        value, ok := counter[key]
        if ok && value == 1 {
            if kth == k {
                return key
            }
            kth++
        }
    }
    return "";
}